src/CategorySource.php

```diff
@@ -0,0 +1,52 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Yiisoft\TranslatorExtractor;
+
+use RuntimeException;
+use Yiisoft\Translator\MessageReaderInterface;
+use Yiisoft\Translator\MessageWriterInterface;
+
+/**
+ * Represents message category for read and write messages
+ */
+final class CategorySource
```

Comment:
Consider using `CategorySource` directly by adding proxy-methods instead of exposing its internals.